### PR TITLE
Fix a bunch of GitHub Actions deprecations

### PR DESCRIPTION
**Pull request by cbjeukendrup**
See:
- https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/
- https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
    - https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter

### CODE REVIEW

**Comment by cbjeukendrup** [comment]
This fixes all things that can be fixed on our side; for the rest, the "sub-actions" that we use will have to be updated. They will probably fix it in 0.0.1 versions, which will get used automatically because we only specify the major version `v3` for the relevant actions.

But anyway: to be on the safe side, **let's not merge this PR until after 4.0** (even though the risk is low because it only involves changes to very basic CI actions that hardly interfere with the build process, and thus have much lower risk than updating a compiler for example). I'm just putting it here so that we have it ready when they suddenly "hard-deprecate" the things.

**Review by igorkorsukov — APPROVED** [review]

**Comment by cbjeukendrup** [comment]
Rebased!

**Review by igorkorsukov — APPROVED** [review]

**Pull request merged**
